### PR TITLE
Fix: Query editor is blank for existing queries on a marketplace plugin data source

### PR DESCRIPTION
**Pull request by devanshu052000**
Closes - https://github.com/ToolJet/tj-ee/issues/5433

This pull request improves the logic for choosing which query editor component to render in the query manager, ensuring correct handling of marketplace plugins, built-in sources, and dummy data sources. The main change is the introduction of a new utility function, `resolveQueryEditor`, which centralizes and clarifies this decision logic. Comprehensive unit tests are also added to prevent regressions and handle edge cases.

**Query Editor Selection Logic Improvements:**

* Added a new `resolveQueryEditor` utility in `utils.js` to determine which editor component should be rendered based on the selected data source and query, addressing previous issues with plugin and built-in editor routing.
* Updated `QueryManagerBody.jsx` to use `resolveQueryEditor` for selecting the editor component, improving reliability and maintainability. [[1]](diffhunk://#diff-0255f10990a502f4bccec18c3b8212b4d4b7562dbfa0e4485042d5d0e05562a9R8) [[2]](diffhunk://#diff-0255f10990a502f4bccec18c3b8212b4d4b7562dbfa0e4485042d5d0e05562a9L56-R61)

**Bug Fixes and Edge Case Handling:**

* Fixed a bug where queries associated with marketplace plugins would not render the correct editor after a reload, by ensuring the editor selection does not rely solely on `plugin_id`. [[1]](diffhunk://#diff-0255f10990a502f4bccec18c3b8212b4d4b7562dbfa0e4485042d5d0e05562a9L56-R61) [[2]](diffhunk://#diff-cf9f53a91b1768f1fa361ed5aeb6d5048287ad1be72b3c61838825f60ec255a6R1-R23)
* Ensured that dummy data sources render no editor to avoid crashes, and that built-in sources always use the correct built-in editor component. [[1]](diffhunk://#diff-0255f10990a502f4bccec18c3b8212b4d4b7562dbfa0e4485042d5d0e05562a9L56-R61) [[2]](diffhunk://#diff-cf9f53a91b1768f1fa361ed5aeb6d5048287ad1be72b3c61838825f60ec255a6R1-R23)

**Testing:**

* Added comprehensive unit tests for `resolveQueryEditor` covering marketplace plugins, built-in sources, default stubs, unresolved sources, and dummy data sources, preventing regressions and clarifying expected behavior.

**Other Minor Improvements:**

* Simplified the logic for passing the `pluginSchema` prop to the editor component, now always using the data source's schema.

### CODE REVIEW

File: frontend/src/AppBuilder/QueryManager/Components/QueryManagerBody.jsx [hunk 1]
```diff
@@ -249,9 +244,7 @@ export const BaseQueryManagerBody = ({ darkMode, activeTab, renderCopilot = null
           <ElementToRender
             renderCopilot={(props) => renderCopilot?.({ ...props, selectedDataSource })}
             key={selectedQuery?.id}
-            pluginSchema={
-              selectedDataSource?.plugin?.operations_file?.data ?? selectedQuery?.plugin?.operations_file?.data
-            }
+            pluginSchema={selectedDataSource?.plugin?.operations_file?.data}
```
[the remark this answers is not in the document]
Comment (devanshu052000) [reply frontend/src/AppBuilder/QueryManager/Components/QueryManagerBody.jsx:247]: Not applicable here — `selectedDataSource` in the App Builder is always snake_case.

It's populated by `fetchGlobalDataSources` → `getForApp`, which runs an unconditional
`decamelizeKeys()` over the whole list (`data-sources/service.ts:59`). The endpoint that
does preserve camelCase for plugin data sources is `getAll`, but its App Builder call
sites are commented out (`useAppData.js:82,735,1004`) — it only serves the Data sources
settings page now.

Two checks confirm it: line 474 of this same file already reads
`selectedDataSource?.plugin_id` and works in production, and before this PR in-session
marketplace queries rendered fine off `selectedDataSource?.plugin?.operations_file?.data`
— the snake_case operand — since the `selectedQuery` fallback beside it is always
undefined.
